module/zfs/zvol.c

```diff
@@ -622,7 +622,12 @@ zvol_replay_truncate(zvol_state_t *zv, lr_truncate_t *lr, boolean_t byteswap)
 	offset = lr->lr_offset;
 	length = lr->lr_length;
 
+#if defined(_KERNEL)
 	return (dmu_free_long_range(zv->zv_objset, ZVOL_OBJ, offset, length));
+#else
+	return (dmu_free_long_range_impl(zv->zv_objset, zv->zv_dn, offset,
```

Comment:
here, version of lr can be checked, and pass NULL, NULL if version is VERSION_0

Reply:
ok